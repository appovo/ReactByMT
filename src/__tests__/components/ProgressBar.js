import ProgressBar from '../../components/ProgressBar'
import renderer from 'react-test-renderer'

var progressBarRenderer = null

describe('when given width, size and color (TestRenderer)', () => {
  beforeEach(() => {
    progressBarRenderer = renderer.create(
      <ProgressBar percent={33} big={true} color="red" />,
    )
  })
  it('renders properly', () => {
    expect(progressBarRenderer.toJSON()).toMatchSnapshot()
    expect(progressBarRenderer.toJSON().props).toMatchObject({
      className: expect.stringMatching(/progress--color-red/),
    })
    expect(progressBarRenderer.toJSON().props).toMatchObject({
      className: expect.stringMatching(/progress--big/),
    })
    expect(progressBarRenderer.toJSON().children[0].props.style).toMatchObject({
      width: '33%',
    })
  })
  it('renders a div element', () => {
    expect(progressBarRenderer.toJSON().type).toEqual('div')
  })
  it('renders children properly', () => {
    expect(progressBarRenderer.toJSON().children[0].type).toEqual('div')
  })
})
