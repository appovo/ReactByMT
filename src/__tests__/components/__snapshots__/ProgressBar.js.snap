// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when given width, size and color (TestRenderer) renders properly 1`] = `
<div
  className="progress progress--big progress--color-red"
>
  <div
    className="progress__bar"
    style={
      Object {
        "width": "33%",
      }
    }
  />
</div>
`;
